frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/PullBackgroundable.java

```diff
@@ -0,0 +1,211 @@
+            gitRepository,
+            currentRev,
+            beforeLabel,
+            updatedRanges);
+      }
+    }
+  }
+
+  @Nullable
+  private static GitUpdatedRanges deriveGitUpdatedRanges(
+      Project project,
+      GitRepository gitRepository,
+      String remoteBranchName) {
+    GitUpdatedRanges updatedRanges = null;
+    var currentBranch = gitRepository.getCurrentBranch();
+    if (currentBranch != null) {
+      String selectedBranch = StringUtil.trimStart(remoteBranchName, /* prefix */ "remotes/");
+      GitBranch targetBranch = gitRepository.getBranches().findBranchByName(selectedBranch);
+      if (targetBranch != null) {
+        GitBranchPair refPair = new GitBranchPair(currentBranch, targetBranch);
+        updatedRanges = GitUpdatedRanges.calcInitialPositions(project,
+            java.util.Collections.singletonMap(gitRepository, refPair));
+      } else {
+        LOG.warn("Couldn't find the branch with name [" + selectedBranch + "]");
+      }
+    }
+    return updatedRanges;
+  }
+
+  private static void handleResult(
+      GitCommandResult result,
+      Project project,
+      GitLocalChangesWouldBeOverwrittenDetector localChangesDetector,
+      GitUntrackedFilesOverwrittenByOperationDetector untrackedFilesDetector,
+      GitRepository repository,
+      GitRevisionNumber currentRev,
+      Label beforeLabel,
+      @Nullable GitUpdatedRanges updatedRanges) {
+    VirtualFile root = repository.getRoot();
+    if (result.success()) {
+      VfsUtil.markDirtyAndRefresh(/* async */ false, /* recursive */ true, /* reloadChildren */ false, root);
+      repository.update();
+      if (updatedRanges != null &&
+          AbstractCommonUpdateAction
+              .showsCustomNotification(java.util.Collections.singletonList(GitVcs.getInstance(project)))) {
+        var ranges = updatedRanges.calcCurrentPositions();
+        GitUpdateInfoAsLog.NotificationData notificationData = new GitUpdateInfoAsLog(project, ranges)
+            .calculateDataAndCreateLogTab();
+
+        Notification notification;
+        if (notificationData != null) {
+          String title = getTitleForUpdateNotification(notificationData.getUpdatedFilesCount(),
+              notificationData.getReceivedCommitsCount());
+          String content = getBodyForUpdateNotification(notificationData.getFilteredCommitsCount());
+          notification = VcsNotifier.STANDARD_NOTIFICATION.createNotification(title,
+              content,
+              INFORMATION,
+              /* listener */ null);
+          notification.addAction(NotificationAction.createSimple(GitMacheteBundle.messagePointer(
+              "GitMachete.PullBackgroundable.notification.message.view-commits"),
+              notificationData.getViewCommitAction()));
+
+        } else {
+          notification = VcsNotifier.STANDARD_NOTIFICATION.createNotification(
+              GitMacheteBundle.message("GitMachete.PullBackgroundable.notification.all-files-are-up-to-date"),
```

Comment:
It's unclear to me why this specific message shows up in this context :thinking:
Maybe some extra comments would be needed around here